src/search/FindInFilesUI.js

```diff
@@ -102,10 +107,45 @@ define(function (require, exports, module) {
 
                 StatusBar.hideBusyIndicator();
             })
+            .progress(function (results) {
+                // Update first page of panel (100 matches) for every file with matches,
+                // then only update the summary for every page (100 matches)
+                if (results.matches && results.matches.length) {
+                    progressCount += results.matches.length;
+                    if (!progressFirstUpdate) {
+                        // First update
+                        _resultsView.open();
+                        if (_findBar) {
+                            _findBar.close();
+                        }
+                        progressFirstUpdate = true;
+                    } else if (!progressFirstPage) {
+                        // First page
+                        if (progressCount >= 100) {
+                            progressCount -= 100;
+                            progressFirstPage = true;
+                        }
+                        _resultsView._updateResults();
+                    } else {
+                        // Subsequent full pages
+                        if (progressCount >= 100) {
+                            progressCount -= 100;
+                            _resultsView._showSummary();
```

Comment:
Since the files are now sorted before searching, and since we are only updating every 100 results there shouldn't be any problem. The idea is that after I press "next page" or "last page" the results shown in that page are complete and won't change when searching over more files.

The point is that the first 100 results go in the first page, the next 100 results in the following and so on. So if I watching the results from 301-400 and the search is over 1000, the next results will go on the page 1001-1100 and nothing on the page I am on right now.

_updateResults() basically just renders the panel and then restores the scroll position and item selected. So it is just needed when updating the page that the user is watching, which we shouldn't need to do. But I am not sure if we can assume the same if we try to process x files in parallel instead of doing all sequentially.